src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -45,6 +45,26 @@ public void storeMessage(Message message) {
     datastore.put(messageEntity);
   }
 
+    /** Stores FoodItem in Datastore. */
+  public void storeFood(FoodItem food) {
+    Entity foodentity = new Entity("FoodItem", food.getID().toString());
+    foodentity.setProperty("name", food.getName());
+    foodentity.setProperty("CO2", food.getCO2());
+    datastore.put(foodentity);
+
+  }
+
+  /** Stores Meal in Datastore. */
+  public void storeMeal(UserMeal meal) {
+    Entity mealEntity = new Entity("UserMeal", meal.getMealID().toString());
+    mealEntity.setProperty("Foods_and_Amounts", meal.getfood_amount());
```

Comment:
`Foods_and_Amounts` is a bit of a strange field name. I would use the same field names that you use on the getter so that you know what it should be right away. So in this case I would switch it to be `foodAmount`.